tensorflow-framework/src/main/java/org/tensorflow/framework/initializers/RandomNormal.java

```diff
@@ -86,16 +85,10 @@ public RandomNormal(Ops tf, double mean, double stddev, long seed) {
 
   /** {@inheritDoc} */
   @Override
-  public <T extends TType> Operand<T> call(Operand<TInt64> dims, Class<T> type) {
-    if (!TNumber.class.isAssignableFrom(type)) {
-      throw new IllegalArgumentException("Tensor type must be numeric: " + type.getSimpleName());
-    }
+  public <U extends TFloating> Operand<U> call(Operand<TInt64> dims, Class<U> type) {
     long[] seeds = {seed, 0};
-    @SuppressWarnings("unchecked")
-    Class<TNumber> nType = (Class<TNumber>) type;
-    Operand<TNumber> distOp = tf.random.statelessRandomNormal(dims, tf.constant(seeds), nType);
-    Operand<TNumber> op =
-        tf.math.mul(distOp, tf.dtypes.cast(tf.constant(this.stddev), distOp.type()));
-    return cast(tf, tf.math.add(op, tf.dtypes.cast(tf.constant(mean), distOp.type())), type);
+    Operand<U> distOp = tf.random.statelessRandomNormal(dims, tf.constant(seeds), type);
+    Operand<U> op = tf.math.mul(distOp, cast(tf, tf.constant(this.stddev), type));
```

Comment:
I missed this initially, but is there a reason we don't check that the standard deviation is positive in the constructor?